holidays/countries/bulgaria.py

```diff
@@ -110,19 +123,14 @@ def _populate_public_holidays(self):
         dts_observed.add(self._add_christmas_day(name))
         dts_observed.add(self._add_christmas_day_two(name))
 
-        if self.observed and self._year >= 2017:
-            excluded_names = {self.tr("Велика събота"), self.tr("Великден")}
-            for dt in sorted(dts_observed):
-                if not self._is_weekend(dt):
-                    continue
-                dt_observed = dt + td(days=+2 if self._is_saturday(dt) else +1)
-                while dt_observed in self:
-                    dt_observed += td(days=+1)
-                for name in self.get_list(dt):
-                    if name not in excluded_names:
-                        self._add_holiday(self.tr("%s (почивен ден)") % name, dt_observed)
+        if self.observed:
+            self.excluded_names = {self.tr("Велика събота"), self.tr("Великден")}
+            self._populate_observed(dts_observed)
```

Comment:
It doesn't need to be an instance variable. Just pass it to the `_populate_observed` (preferably inlined):

```suggestion
            self._populate_observed(
                dts_observed, excluded_names={self.tr("Велика събота"), self.tr("Великден")}
            )
```

If the holiday name based exclusion is going to be used somewhere else too I'd suggest making `excluded_names` logic a part of `ObservedHolidayBase`.

Reply:
Done.